dev/TeachingTip/TeachingTip.cpp

```diff
@@ -1470,68 +1495,75 @@ std::tuple<winrt::TeachingTipPlacementMode, bool> TeachingTip::DetermineEffectiv
         if (clippedTargetBounds.Left < -targetBounds.Width / 2 ||
             clippedTargetBounds.Right < -targetBounds.Width / 2)
         {
-            availability[winrt::TeachingTipPlacementMode::TopEdgeAlignedLeft] = false;
+            availability[winrt::TeachingTipPlacementMode::TopLeft] = false;
             availability[winrt::TeachingTipPlacementMode::Top] = false;
-            availability[winrt::TeachingTipPlacementMode::TopEdgeAlignedRight] = false;
-            availability[winrt::TeachingTipPlacementMode::BottomEdgeAlignedLeft] = false;
+            availability[winrt::TeachingTipPlacementMode::TopRight] = false;
+            availability[winrt::TeachingTipPlacementMode::BottomLeft] = false;
             availability[winrt::TeachingTipPlacementMode::Bottom] = false;
-            availability[winrt::TeachingTipPlacementMode::BottomEdgeAlignedRight] = false;
+            availability[winrt::TeachingTipPlacementMode::BottomRight] = false;
+			availability[winrt::TeachingTipPlacementMode::Center] = false;
         }
 
         // If the vertical midpoint is out of the window.
         if (clippedTargetBounds.Top < -targetBounds.Height / 2 ||
             clippedTargetBounds.Bottom < -targetBounds.Height / 2)
         {
-            availability[winrt::TeachingTipPlacementMode::LeftEdgeAlignedBottom] = false;
+            availability[winrt::TeachingTipPlacementMode::LeftBottom] = false;
             availability[winrt::TeachingTipPlacementMode::Left] = false;
-            availability[winrt::TeachingTipPlacementMode::LeftEdgeAlignedTop] = false;
-            availability[winrt::TeachingTipPlacementMode::RightEdgeAlignedBottom] = false;
+            availability[winrt::TeachingTipPlacementMode::LeftTop] = false;
+            availability[winrt::TeachingTipPlacementMode::RightBottom] = false;
             availability[winrt::TeachingTipPlacementMode::Right] = false;
-            availability[winrt::TeachingTipPlacementMode::RightEdgeAlignedTop] = false;
+            availability[winrt::TeachingTipPlacementMode::RightTop] = false;
+			availability[winrt::TeachingTipPlacementMode::Center] = false;
         }
 
         // If the tip is too tall to fit between the top of the target and the top edge of the window or screen.
         if (tipHeight > availableBoundsAroundTarget.Top)
         {
             availability[winrt::TeachingTipPlacementMode::Top] = false;
-            availability[winrt::TeachingTipPlacementMode::TopEdgeAlignedRight] = false;
-            availability[winrt::TeachingTipPlacementMode::TopEdgeAlignedLeft] = false;
-        }
+            availability[winrt::TeachingTipPlacementMode::TopRight] = false;
+            availability[winrt::TeachingTipPlacementMode::TopLeft] = false;
+        }
+		// If the total tip is too tall to fit between the center of the target and the top of the window.
+		if (tipHeight > availableBoundsAroundTarget.Top + (targetBounds.Height / 2.0f))
```

Comment:
I don't understand this check. I thought in Center placement the tip is centered in the target. So I would have expected a check against targetBounds.Height + availableBoundsAroundTarget.Top + availableBoundsAroundTarget.Bottom -- like only if we are going to fall out of the window on both top/bottom would we rule it out.

Also is there any X/width constraint that rule out center placement?

Reply:
Center placed tips have the content placed on top of the pointer which points to the center of the target.  This means that the content of the tip will never be below the center of the target so we don't have to worry about the tip being bottom clipped (since we already checked to make sure the center of the target was visible).

Yes, we should add center to the list of invalidated zones during the center-center check below, thanks for catching.